.github/workflows/test-linux.yaml

```diff
@@ -133,15 +133,27 @@ jobs:
     name: linux-cp313t-default
     runs-on: ubuntu-latest
 
+    strategy:
+      # Ensure that a wheel builder finishes even if another fails
+      fail-fast: false
+      matrix:
+        run_parallel: [0, 1]
+        version: ['3.13t']
+    env:
+      RUN_PARALLEL: ${{ matrix.run_parallel }}
+
     steps:
       - name: Checkout scikit-image
         uses: actions/checkout@v4
 
       # TODO: replace with setup-python when there is support
-      - uses: deadsnakes/action@6c8b9b82fe0b4344f4b98f2775fcc395df45e494 # v3.1.0
+      # TODO: replace with setup-python when there is support
+      - uses: astral-sh/setup-uv@b5f58b2abc5763ade55e4e9d0fe52cd1ff7979ca
         with:
-          python-version: "3.13"
-          nogil: true
+          python-version: ${{ matrix.version }}
+          enable-cache: false
```

Comment:
Any reason not to use the cache here?

Reply:
Perhaps cribbed from NumPy's config, where I intentionally didn't use the github cache because NumPy's cache is already too big.

Reply:
Oh, okay. Might be a dumb question, but what qualifies it as "too big"? Is there a limit imposed by GitHub or is it detrimental in another way?

Reply:
Each repo gets 10 GB of github actions cache space to share across all actions jobs. If you go over the limit, github evicts the least recently used cache item. 

In practice on the NumPy CI that means if we go over the limit, caches randomly need to be repopulated in a round-robin fashion depending on the last thing to be evicted from the cache, making CI runtimes unpredictable.